app/src/main/java/com/example/soccert/ui/news/NewsFragment.kt

```diff
@@ -24,15 +33,47 @@ class NewsFragment : BaseFragment<FragmentNewsBinding>() {
     }
 
     override fun initData() {
-
+        binding.apply {
+            lifecycleOwner = viewLifecycleOwner
+            newsVM = viewModel
+            recyclerNews.adapter = newsAdapter
+        }
     }
 
     override fun initActions() {
-
+        viewModel.error.observe(viewLifecycleOwner, Observer {
+            context?.showToast(ToastType.Error, it)
+        })
     }
 
     override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
         inflater.inflate(R.menu.menu_search, menu)
         super.onCreateOptionsMenu(menu, inflater)
     }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.text_menu_search -> {
+                findNews(item)
+            }
```

Comment:
```suggestion
            R.id.text_menu_search ->  findNews(item)
```
nếu gọn trong 1 hàm r thì viêt liền thế này luôn

Reply:
vâng a <like